### ISSUE TITLE
Terraform plan crashes with aws_autoscalingplans_scaling_plan resource with customized_scaling_metric_specification block

### ISSUE THREAD

**Issue by ckilpatrick20** [hashicorp/terraform#27950]
### Terraform Version

```
v0.12.29
```

### Terraform Configuration Files

```terraform
resource "aws_launch_template" "this" {
  name                                 = var.lt_name
  description                          = var.lt_description
  update_default_version               = var.update_default_version
  ebs_optimized                        = var.ebs_optimized
  image_id                             = var.image_id
  instance_type                        = var.instance_type
  instance_initiated_shutdown_behavior = var.instance_initiated_shutdown_behavior

  dynamic "iam_instance_profile" {
    for_each = var.iam_instance_profile != null ? [1] : []
    content {
      name = var.iam_instance_profile
    }
  }

  key_name               = var.key_name
  vpc_security_group_ids = var.associate_public_ip_address == null ? var.vpc_security_group_ids : null
  user_data              = var.user_data

  dynamic "network_interfaces" {
    for_each = var.associate_public_ip_address != null ? [1] : []
    content {
      associate_public_ip_address = var.associate_public_ip_address
      security_groups             = var.vpc_security_group_ids
    }
  }

  monitoring {
    enabled = var.enable_monitoring
  }

  lifecycle {
    create_before_destroy = true
  }
}

resource "aws_autoscaling_group" "this" {
  name = var.asg_name

  launch_template {
    id      = aws_launch_template.this.id
    version = var.lt_version != null ? var.lt_version : "$Latest"
  }

  vpc_zone_identifier = var.vpc_zone_identifier
  max_size            = var.max_size
  min_size            = var.min_size

  health_check_grace_period = var.health_check_grace_period
  health_check_type         = var.health_check_type

  min_elb_capacity          = var.min_elb_capacity
  wait_for_elb_capacity     = var.wait_for_elb_capacity
  target_group_arns         = var.target_group_arns
  default_cooldown          = var.default_cooldown
  force_delete              = var.force_delete
  termination_policies      = var.termination_policies
  suspended_processes       = var.suspended_processes
  enabled_metrics           = var.enabled_metrics
  wait_for_capacity_timeout = var.wait_for_capacity_timeout
  protect_from_scale_in     = var.protect_from_scale_in
  service_linked_role_arn   = var.service_linked_role_arn

  dynamic "instance_refresh" {
    for_each = var.instance_refresh
    content {
      strategy = "Rolling"
      preferences {
        min_healthy_percentage = instance_refresh.value["min_healthy_percentage"]
      }
    }
  }

  tags = concat(
    [
      {
        "key"                 = "Name"
        "value"               = "${var.asg_name}"
        "propagate_at_launch" = true
      },
    ],
    local.tags_asg_format,
  )

  lifecycle {
    create_before_destroy = true
  }
}

resource "aws_autoscalingplans_scaling_plan" "this" {
  name = var.asp_name

  application_source {
    tag_filter {
      key    = "Name"
      values = [var.asg_name]
    }
  }

  scaling_instruction {
    max_capacity       = var.max_size
    min_capacity       = var.min_size
    resource_id        = format("autoScalingGroup/%s", aws_autoscaling_group.this.name)
    scalable_dimension = "autoscaling:autoScalingGroup:DesiredCapacity"
    service_namespace  = "autoscaling"

    target_tracking_configuration {
      customized_scaling_metric_specification {
        metric_name = "Memory % Committed Bytes In Use"
        namespace   = "web_tier"
        statistic   = "Average"
        dimensions = {
          "AutoScalingGroupName" = "web_asg"
        }
      }

      target_value = var.target_value
    }

    target_tracking_configuration {
      predefined_scaling_metric_specification {
        predefined_scaling_metric_type = var.predefined_scaling_metric_type
      }

      target_value = var.target_value
    }
  }
}
```

### Crash Output

```
panic: inconsistent set element types (cty.Object(map[string]cty.Type{"customized_scaling_metric_specification":cty.Tuple([]cty.Type{cty.Object(map[string]cty.Type{"dimensions":cty.Map(cty.String), "metric_name":cty.String, "namespace":cty.String, "statistic":cty.String, "unit":cty.String})}), "disable_scale_in":cty.Bool, "estimated_instance_warmup":cty.Number, "predefined_scaling_metric_specification":cty.EmptyTuple, "scale_in_cooldown":cty.Number, "scale_out_cooldown":cty.Number, "target_value":cty.Number}) then cty.Object(map[string]cty.Type{"customized_scaling_metric_specification":cty.EmptyTuple, "disable_scale_in":cty.Bool, "estimated_instance_warmup":cty.Number, "predefined_scaling_metric_specification":cty.Tuple([]cty.Type{cty.Object(map[string]cty.Type{"predefined_scaling_metric_type":cty.String, "resource_label":cty.String})}), "scale_in_cooldown":cty.Number, "scale_out_cooldown":cty.Number, "target_value":cty.Number}))

goroutine 1097 [running]:
github.com/zclconf/go-cty/cty.SetVal(0xc001b052c0, 0x2, 0x2, 0x27cf4a0, 0xc00247e150, 0x3068e00, 0x3068e40)
	/go/pkg/mod/github.com/zclconf/go-cty@v1.2.1/cty/value_init.go:258 +0x76d
github.com/hashicorp/terraform/plans/objchange.setElementCompareValue(0xc000c34390, 0x3068e40, 0xc001792768, 0x27cf4a0, 0xc00247faa0, 0x3068e00, 0xc001792768, 0x27cf4a0, 0xc00247faa0, 0x13)
	/home/circleci/project/project/plans/objchange/objchange.go:358 +0x1149
github.com/hashicorp/terraform/plans/objchange.setElementCompareValues(0xc000c34390, 0x3068e80, 0xc0027183f0, 0x2a714a0, 0xc0007e7d00, 0x0, 0x3068e80, 0xc000dee1e0, 0x2a714a0)
	/home/circleci/project/project/plans/objchange/objchange.go:300 +0x1b3
github.com/hashicorp/terraform/plans/objchange.proposedNewObject(0xc000c1c530, 0x3068e40, 0xc0017927b8, 0x27cf4a0, 0xc002483680, 0x3068e40, 0xc0017933a0, 0x27cf4a0, 0xc001aadc50, 0x20, ...)
	/home/circleci/project/project/plans/objchange/objchange.go:237 +0x1892
github.com/hashicorp/terraform/plans/objchange.ProposedNewObject(0xc000c1c530, 0x3068e40, 0xc0017927b8, 0x27cf4a0, 0xc002483680, 0x3068e40, 0xc0017933a0, 0x27cf4a0, 0xc001aadc50, 0x0, ...)
	/home/circleci/project/project/plans/objchange/objchange.go:42 +0x127
github.com/hashicorp/terraform/terraform.(*EvalDiff).Eval(0xc0010142a0, 0x309f040, 0xc000cd4f70, 0x2, 0x2, 0x0, 0x0)
	/home/circleci/project/project/terraform/eval_diff.go:164 +0x6d6
github.com/hashicorp/terraform/terraform.EvalRaw(0x30235e0, 0xc0010142a0, 0x309f040, 0xc000cd4f70, 0x0, 0x0, 0x0, 0x0)
	/home/circleci/project/project/terraform/eval.go:57 +0x131
github.com/hashicorp/terraform/terraform.(*EvalSequence).Eval(0xc0007d80a0, 0x309f040, 0xc000cd4f70, 0x2, 0x2, 0x1b3db95, 0x3023a00)
	/home/circleci/project/project/terraform/eval_sequence.go:20 +0xfd
github.com/hashicorp/terraform/terraform.EvalRaw(0x3023920, 0xc0007d80a0, 0x309f040, 0xc000cd4f70, 0x27bfa80, 0x4134a62, 0x27231c0, 0xc00133e5a0)
	/home/circleci/project/project/terraform/eval.go:57 +0x131
github.com/hashicorp/terraform/terraform.Eval(0x3023920, 0xc0007d80a0, 0x309f040, 0xc000cd4f70, 0xc0007d80a0, 0x3023920, 0xc0007d80a0, 0x0)
	/home/circleci/project/project/terraform/eval.go:35 +0x4d
github.com/hashicorp/terraform/terraform.(*Graph).walk.func1(0x2a74500, 0xc000bc4730, 0x0, 0x0, 0x0)
	/home/circleci/project/project/terraform/graph.go:90 +0xf40
github.com/hashicorp/terraform/dag.(*Walker).walkVertex(0xc000458580, 0x2a74500, 0xc000bc4730, 0xc0006f6700)
	/home/circleci/project/project/dag/walk.go:392 +0x353
created by github.com/hashicorp/terraform/dag.(*Walker).Update
	/home/circleci/project/project/dag/walk.go:314 +0xa9b



!!!!!!!!!!!!!!!!!!!!!!!!!!! TERRAFORM CRASH !!!!!!!!!!!!!!!!!!!!!!!!!!!!

Terraform crashed! This is always indicative of a bug within Terraform.
A crash log has been placed at "crash.log" relative to your current
working directory. It would be immensely helpful if you could please
report the crash with Terraform[1] so that we can fix this.

When reporting bugs, please include your terraform version. That
information is available on the first line of crash.log. You can also
get it by running 'terraform --version' on the command line.

SECURITY WARNING: the "crash.log" file that was created may contain
sensitive information that must be redacted before it is safe to share
on the issue tracker.

[1]: https://github.com/hashicorp/terraform/issues

!!!!!!!!!!!!!!!!!!!!!!!!!!! TERRAFORM CRASH !!!!!!!!!!!!!!!!!!!!!!!!!!!!
```

### Expected Behavior

```
terraform apply
...
Apply complete! Resources: 3 added, 0 changed, 0 destroyed.
Releasing state lock. This may take a few moments...

terraform plan
...
No changes. Infrastructure is up-to-date.
```

### Actual Behavior

Terraform plan crashes with panic output shown in above crash output

### Steps to Reproduce

1. `terraform apply`
2. `terraform plan`

### Additional Context

The steps listed above work if I omit the the block of code below in the aws_autoscalingplans_scaling_plan resource. Once added, the apply will create the resources but when followed by a plan, Terraform will crash.
```
  target_tracking_configuration {
    customized_scaling_metric_specification {
      metric_name = "Memory % Committed Bytes In Use"
      namespace   = "web_tier"
      statistic   = "Average"
      dimensions = {
        "AutoScalingGroupName" = "web_asg"
      }
    }

    target_value = var.target_value
  }
```

**Comment by jbardin** [comment]
Hi @ckilpatrick20 

Thanks for filing the issue. The configuration shown is missing a lot of variable values that would be needed to use it as a reproduction case. Can you verify that this is still an issue with a current Terraform release? If the issue still exists we can work on making this a standalone reproduction case.

Thanks!

**Comment by ckilpatrick20** [comment]
Hi @jbardin 

After upgrading to version 14.7 I can confirm this issue is fixed. It appears it was fixed in a release between 13.6 and 14.7. 

Thank you for your time.

**Comment by jbardin** [comment]
Thanks for the update @ckilpatrick20!
